internal/services/postgres/postgresql_server_resource.go

```diff
@@ -663,21 +697,41 @@ func resourcePostgreSQLServerUpdate(d *pluginsdk.ResourceData, meta interface{})
 	}
 
 	// Update Admin Password in the separate call when Replication is stopped: https://github.com/Azure/azure-rest-api-specs/issues/16898
-	if d.HasChange("administrator_login_password") && !replicaUpdatedToDefault {
-		properties.Properties.AdministratorLoginPassword = utils.String(d.Get("administrator_login_password").(string))
+	if (d.HasChange("administrator_login_password") || d.HasChange("administrator_login_password_version")) && !replicaUpdatedToDefault {
+		password := ""
+
+		if v, ok := d.GetOk("administrator_login_password"); ok {
+			password = v.(string)
+		}
+		woPassword, err := pluginsdk.GetWriteOnly(d, "administrator_login_password_wo", cty.String)
+		if err != nil {
+			return err
+		}
+		if !woPassword.IsNull() {
+			password = woPassword.AsString()
+		}
+		properties.Properties.AdministratorLoginPassword = pointer.To(password)
 	}
 
 	if err = client.UpdateThenPoll(ctx, *id, properties); err != nil {
 		return fmt.Errorf("updating %s: %+v", *id, err)
 	}
 
 	// Update Admin Password in a separate call when Replication is stopped: https://github.com/Azure/azure-rest-api-specs/issues/16898
-	if d.HasChange("administrator_login_password") && replicaUpdatedToDefault {
-		properties.Properties.AdministratorLoginPassword = utils.String(d.Get("administrator_login_password").(string))
+	if (d.HasChange("administrator_login_password") || d.HasChange("administrator_login_password_wo")) && replicaUpdatedToDefault {
```

Comment:
Same question regarding `HasChange` vs `HasChanges` as above and I think `administrator_login_password_wo` should be `administrator_login_password_wo_version`

```suggestion
	if d.HasChanges("administrator_login_password", "administrator_login_password_wo_version") && replicaUpdatedToDefault {
```